camel-chimera-graph/src/main/java/com/cefriel/operations/GraphSparql.java

```diff
@@ -0,0 +1,93 @@
+import org.eclipse.rdf4j.query.resultio.text.csv.SPARQLResultsCSVWriter;
+import org.eclipse.rdf4j.query.resultio.text.tsv.SPARQLResultsTSVWriter;
+import org.eclipse.rdf4j.repository.Repository;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+public class GraphSparql {
+    private static final Logger LOG = LoggerFactory.getLogger(GraphSparql.class);
+    private static final Set<String> validOutputFormats = Set.of("json", "csv", "xml", "tsv", "memory");
+    private record EndpointParams (String literalQuery, ChimeraResourceBean resourceQuery, String outputFormat) {
+        EndpointParams(GraphBean operationConfig) {
+            this(operationConfig.getQuery(),
+                    operationConfig.getChimeraResource(),
+                    operationConfig.getDumpFormat());
+        }
+        EndpointParams {
+            Objects.requireNonNull(outputFormat, "format for sparql query result must be specified using dumpFormat endpoint option");
+
+            if (!validOutputFormats.contains(outputFormat))
+                throw new IllegalArgumentException("unsupported format " + outputFormat + ", must be one of: " + validOutputFormats);
+
+            if (literalQuery == null&& resourceQuery == null)
+                throw new NullPointerException("both sparql queries query cannot be null");
+        }
+    }
+    private record OperationParams (Exchange exchange, RDFGraph graph, String query, String outputFormat) {
+        OperationParams(Exchange exchange, EndpointParams endpointParams) throws Exception {
+            this(exchange,
+                    exchange.getMessage().getBody(RDFGraph.class),
+                    Utils.resolveQuery(endpointParams.literalQuery, endpointParams.resourceQuery, exchange),
+                    endpointParams.outputFormat);
+        }
+
+        OperationParams {
+            Objects.requireNonNull(exchange, "Incoming exchange cannot be null");
+            Objects.requireNonNull(graph, "RDFGraph cannot be null");
+
+            if (query.isBlank())
+                throw new IllegalArgumentException("SPARQL query cannot be blank");
+        }
+    }
+    public static void graphSparql(Exchange exchange, GraphBean operationConfig) throws Exception {
+        EndpointParams endpointParams = new EndpointParams(operationConfig);
+        OperationParams operationParams = new OperationParams(exchange, endpointParams);
+        graphSparql(exchange, operationParams.graph, operationParams.query, operationParams.outputFormat);
+    }
+    private static void graphSparql (Exchange exchange, RDFGraph graph, String query, String outputFormat) {
+        Repository repo = graph.getRepository();
+        RepositoryConnection connection = repo.getConnection();
+        TupleQuery tupleQuery = connection.prepareTupleQuery(query);
```

Comment:
Because of how RDF4J handles different types of sparql queries this is automatically handled by RDF4J. If the user submits the wrong type of query, e.g, a CONSTRUCT query to GraphSparql then an exception is thrown and the user is informed of the mistake.